### PR TITLE
Add support for nvdimm device

### PR DESCRIPTION
**Pull request by huangyum**
Add support for nvdimm device
ID: 1400770

Signed-off-by: Yumei Huang <yuhuang@redhat.com>

### CODE REVIEW

File: virttest/qemu_devices/qdevices.py [hunk 1]
```diff
@@ -927,8 +927,22 @@ def verify_hotplug(self, out, monitor):
         return False
 
 
-class Dimm(QDevice):
+class Nvdimm(QDevice):
```
Comment (xutian) [inline virttest/qemu_devices/qdevices.py]: can you have a try about hotplug/unplug function?

File: virttest/qemu_devices/qdevices.py [hunk 2]
```diff
@@ -927,8 +927,22 @@ def verify_hotplug(self, out, monitor):
...
+    """
+    NVDIMM device, using the QOM objects 'memory-backend=file', not
+    support hotplug yet.
+    """
+    __attributes__ = ["memdev", "slot", "addr", "node"]
+
+    def __init__(self, params=None):
```
Comment (xutian) [inline virttest/qemu_devices/qdevices.py]: view from your code, nvdimm should another kind of dimm device, so I think better ways is redfine __init__() for DIMM devices.

File: virttest/qemu_devices/qdevices.py (hunk 1 above)
Comment (huangyum) [inline virttest/qemu_devices/qdevices.py]: I didn't remove the Dimm class, just add a new Nvdimm class, which will not affect hotplug/unplug function.  Anyway, the test for hotplug passed.  
(1/1) smp_16.4096m.repeat1.run_test.Host_RHEL.m7.u3.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.providers.hotplug_memory.before.vm_system_reset.hotplug.backend_file.2M_hugepage.policy_default.two: PASS (157.45 s)

Comment (xutian) [inline virttest/qemu_devices/qdevices.py]: I mean nvdimm hotplug/unplug

Comment (xutian) [inline virttest/qemu_devices/qdevices.py]: And I think most of testcase for dimm device can be reused for nvdimm device.

Comment (huangyum) [inline virttest/qemu_devices/qdevices.py]: Nvidmm hotplug/unplug is not supported yet. I reused the dimm class, and will update the hotplug function when it is supported. Please help review again. Thanks!

File: virttest/qemu_devices/qcontainer.py [hunk 3]
```diff
@@ -1794,9 +1794,12 @@ def dimm_device_define_by_params(self, params, name):
         Create pc-dimm device from params.
         """
         params = params.object_params("dimm")
+        dimm_type = 'pc-dimm'
+        if params.get('share'):
+            dimm_type = 'nvdimm'
```
Comment (xutian) [inline virttest/qemu_devices/qcontainer.py]: dimm_type = "share" in params and  ["nvdimm"] or  "pc-dimm"

Comment (huangyum) [inline virttest/qemu_devices/qcontainer.py]: Thanks， updated!

**Comment by xutian** [comment]
Should be device or test, please update your summary

**Comment by huangyum** [comment]
Thanks @vivianQizhu  @xutian , updated.

**Pull request merged**